src/main/java/io/github/nucleuspowered/nucleus/configurate/typeserialisers/DelayedTimeValueSerialiser.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.configurate.typeserialisers;
+
+import com.google.common.reflect.TypeToken;
+import com.cronutils.model.Cron;
+import com.cronutils.model.CronType;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.parser.CronParser;
+import io.github.nucleuspowered.nucleus.util.ClockTime;
+import io.github.nucleuspowered.nucleus.util.DelayedTimeType;
+import io.github.nucleuspowered.nucleus.util.DelayedTimeValue;
+import io.github.nucleuspowered.nucleus.util.WhenTime;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.SimpleConfigurationNode;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import ninja.leaping.configurate.objectmapping.serialize.TypeSerializer;
+
+import java.util.Map;
+
+/**
+ * Manages serialization of a DelayedTimeValue
+ */
+public class DelayedTimeValueSerialiser implements TypeSerializer<DelayedTimeValue> {
+
+    private CronParser unixCronParser;
+    private CronParser quartzCronParser;
+    private CronParser cronjCronParser;
+
+    /**
+     * Constructs a DelayedTimeValueSerialiser, initializes the cron parsers
+     */
+    public DelayedTimeValueSerialiser() {
+        this.unixCronParser = new CronParser(CronDefinitionBuilder.instanceDefinitionFor(CronType.UNIX));
+        this.quartzCronParser = new CronParser(CronDefinitionBuilder.instanceDefinitionFor(CronType.QUARTZ));
+        this.cronjCronParser = new CronParser(CronDefinitionBuilder.instanceDefinitionFor(CronType.CRON4J));
+    }
+
+    @Override
+    public DelayedTimeValue deserialize(TypeToken<?> type, ConfigurationNode value) throws ObjectMappingException {
+        if (value.getValue() instanceof String) {
+            String node = value.getString();
+            try {
```

Comment:
This block of code has effectively been duplicated:

* where this line is (obviously)
* [in your DelayedTimeValueArgument](https://github.com/NucleusPowered/Nucleus/pull/555/files#diff-e5811f0135c71cc7b4ef6aaa39f5a028R66)
* [later in this class, ish!](https://github.com/NucleusPowered/Nucleus/pull/555/files#diff-d0395d2ec02f2867fd00243e1cfae254R93)

which would make it hard (and error prone) to add new serialisers/deserialisers if that ever needed to be the case.

What I suggest you do is make your [DelayedTimeType enum](https://github.com/NucleusPowered/Nucleus/pull/555/files#diff-d9d729bcae0fa034156f3ff089dc1799R12) handle this. If you create an interface for it with, say `String serialise(T) and `Optional<T> deserialise(String)` or `T deserialise(String) throws Exception`, you can then collapse the logic into one place whilst still keeping the link between type and action. Where you're just scanning for the right time type, you can just iterate over the enum values and execute the required method.